std/arithmetic.c3

```diff
@@ -15,12 +15,7 @@ typedef CommonArithmeticType<i32, i128> : i128
 typedef CommonArithmeticType<i64, i128> : i128
 
 typedef [U, V] ArithmeticCompareResult : TypeIf<bool, Both<IsIntegral<U>, IsIntegral<V>>>
-
-typedef HasTrivialAssignments<i8> : TrueType
-typedef HasTrivialAssignments<i16> : TrueType
-typedef HasTrivialAssignments<i32> : TrueType
-typedef HasTrivialAssignments<i64> : TrueType
-typedef HasTrivialAssignments<i128> : TrueType
+typedef [T] RegisterHasTrivialAssignments<Overwrite<T>> : TypeIf<TrueType, IsIntegral<T>>
```

Comment:
What's the purpose of `Overwrite`?

Reply:
Let's say we want to do something like this:
```c++
typedef [T] MyTypeTrait : FalseType // The default case
typedef [T] MyTypeTrait : TypeIf<TrueType, IsIntegral<T>>  // Integral types should all have `MyTypeTrait`
```
`MyTypeTrait<int>` fails with the error `'MyTypeTrait<int>' is ambiguous` since both typedefs pass SFINAE and have the same pattern matching cost.

`Override` (which I now think is a better name than `Overwrite`) gets around this issue by giving the default case a higher pattern maching cost than the specialized versions:
```c++
typedef [T] RegisterMyTypeTrait<Override<T>> : ... // Pattern match `T` in `Override<T>`
typedef [T] RegisterMyTypeTrait<T> : ...  // Pattern match `T` by itself
```

Reply:
Ahhh, nice. That makes sense